### DISCUSSION TITLE
How to do natural sort on a string column

### DISCUSSION THREAD

**Discussion by SharadKumar** [prisma/prisma#3693] [Q&A] [answered]
This one is classic case of sorting on a column (asc/desc) that has string values, but alphanum. I need to sort/paginate on this large table, `order by` this alphanum column (think of it as ecommerce product SKU).

ASC (`12` should be the last one!)
![image](https://user-images.githubusercontent.com/5488144/93547407-5559ec80-f9a8-11ea-9c55-5ac91f7d32a0.png)

DESC (`6` should be the last one!)
![image](https://user-images.githubusercontent.com/5488144/93547499-84705e00-f9a8-11ea-9560-28a11378c9f5.png)

## Question
How to do Order By on below `Product.number` column?

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["connectOrCreate"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Product {
  id          String    @default(cuid()) @id
  number      String? // <<< natural sort this >>>
  title       String?
  price       Float?
  currency    String?
  createdAt   DateTime  @default(now()) @map("created_at")
  updatedAt   DateTime  @map("updated_at") @updatedAt
}
```

**Comment by ryands17** [comment, marked as answer]
Hey @SharadKumar 👋 
Your best bet here would be using `prisma.$queryRaw` as this isn't supported by Prisma currently.

Something like this should work:

```ts
await prisma.$queryRaw`
select * from "Product"
order by 
	(substring("number", '^[0-9]+'))::int, 
	substring("number", '[^0-9_].*$')
`
```

You can use this directly and get the intended result. For descending, you can use the `desc` keyword. Also it would be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for this.

**Reply by SharadKumar** [reply]
Thanks Ryan, let me try and report back.

Meantime, I have created the feature request #3707

**Reply by SharadKumar** [reply]
Okay, it worked like a charm!

![image](https://user-images.githubusercontent.com/5488144/93652379-2733e580-fa58-11ea-850e-41d5fa4f40a1.png)

and DESC

![image](https://user-images.githubusercontent.com/5488144/93652596-e7b9c900-fa58-11ea-829c-f3904261e4b1.png)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
